idflakies-maven-plugin/src/main/java/edu/illinois/cs/dt/tools/plugin/IncDetectorMojo.java

```diff
@@ -0,0 +1,635 @@
+        startsSelectedTestsPath = relativePath(PathManager.startsPath(), Paths.get("selected-tests"));
+        startsDependenciesPath = relativePath(PathManager.startsPath(), Paths.get("deps.zlc"));
+
+        transitiveClosure = new HashMap<>();
+        reverseTransitiveClosure = new HashMap<>();
+
+        getSureFireClassPath(project);
+        loader = createClassLoader(sureFireClassPath);
+
+        return null;
+    }
+
+    private String getArtifactsDir() throws FileNotFoundException {
+        if (artifactsDir == null) {
+            artifactsDir = PathManager.cachePath().toString();
+            File file = new File(artifactsDir);
+            if (!file.mkdirs() && !file.exists()) {
+                throw new FileNotFoundException("I could not create artifacts dir: " + artifactsDir);
+            }
+        }
+        return artifactsDir;
+    }
+
+    private String getStartsDir() throws FileNotFoundException {
+        if (startsDir == null) {
+            startsDir = PathManager.startsPath().toString();
+            File file = new File(startsDir);
+            if (!file.mkdirs() && !file.exists()) {
+                throw new FileNotFoundException("I could not create artifacts dir: " + startsDir);
+            }
+        }
+        return startsDir;
+    }
+
+    private String getEkstaziDir() throws FileNotFoundException {
+        if (ekstaziDir == null) {
+            ekstaziDir = PathManager.ekstaziPath().toString();
+            File file = new File(ekstaziDir);
+            if (!file.mkdirs() && !file.exists()) {
+                throw new FileNotFoundException("I could not create artifacts dir: " + ekstaziDir);
+            }
+        }
+        return ekstaziDir;
+    }
+
+    private List<String> getCleanClassPath(String cp) {
+        List<String> cpPaths = new ArrayList<>();
+        String[] paths = cp.split(File.pathSeparator);
+        String classes = File.separator + TARGET + File.separator + CLASSES;
+        String testClasses = File.separator + TARGET + File.separator + TEST_CLASSES;
+        Set<String> classPathSet = new HashSet<>();
+        for (int i = 0; i < paths.length; i++) {
+            paths[i].replaceAll(" ", "");
+            // TODO: should we also exclude SNAPSHOTS from same project?
+            if (paths[i].contains(classes) || paths[i].contains(testClasses)) {
+                continue;
+            }
+            classPathSet.add(paths[i]);
+        }
+        for(String classPath: classPathSet) {
+            cpPaths.add(classPath);
+        }
+        return cpPaths;
+    }
+
+    private Map<String, Set<String>> getReverseClosure(Map<String, Set<String>> transitiveClosure) {
+        Map<String, Set<String>> reverseTransitiveClosure = new HashMap<>();
+        for (Map.Entry<String, Set<String>> entry : transitiveClosure.entrySet()) {
+            for (String dep : entry.getValue()) {
+                Set<String> reverseDeps = new HashSet<>();
```

Comment:
Only need to create new `HashSet` when you need it, namely in the `else` branch.